OasisDEX/oasis-borrow
# Rz/integrate gnosis

# [Gnosis Wallet Support/Disable Sign Message](https://app.clubhouse.io/oazo-apps/story/1857/on-hold-add-support-for-gnosis-safe)

## Changes 👷‍♀️
- update `web3-context` from common
- add web3-react connector for Gnosis Safe
- detect if user is connecting with Gnosis Safe, via connector or Wallet Connect, if so skip signing message until there is support for offchain signatures
  
## How to test 🧪
- go to https://gnosis-safe.io/app/ and set up your safe
- connect via WalletConnect on preview url
- connect via custom app - `https://gnosis-safe.io/app/#/safes/MY_SAFE/apps?appUrl=https://oasis-borrow-rz-integra-subcem.herokuapp.com/borrow` on preview url
- see that app only asks for ToS and saves that in Local Storage, without asking to sign message
    
## Definition of done ✔️

## Note
- currently contacting Gnosis Safe about the fact that preview url is not detected if we want to add it by `custom app` :(
